src/storage_utils.cpp

```diff
@@ -63,19 +63,20 @@ namespace libtorrent { namespace aux {
 		return ret;
 	}
 
-	typed_span<iovec_t> advance_bufs(typed_span<iovec_t> bufs, int const bytes)
+	typed_span<iovec_t> advance_bufs(typed_span<iovec_t> const& bufs, int const bytes)
 	{
```

Comment:
I would sooner believe there's bug in the debugger than a bug in the compiler, especially if you're debugging optimized code. I would take a look at the assembly output to be sure of what's going on.